src/write_xlsx.c

```diff
@@ -170,7 +165,7 @@ SEXP C_write_data_frame_list(SEXP df_list, SEXP file, SEXP col_names, SEXP forma
         case COL_HYPERLINK:{
           SEXP val = STRING_ELT(col, i);
           if(val != NA_STRING && Rf_length(val))
-            assert_lxw(worksheet_write_formula(sheet, cursor, j, Rf_translateCharUTF8(val), hyperlink));
+            assert_lxw(worksheet_write_url(sheet, cursor, j, Rf_translateCharUTF8(val), NULL));
```

Comment:
This should be `worksheet_write_url_opt()` to also set the string name for the hyperlink. But that is an easy change.